be/src/storage/lake/metadata_iterator.cpp

```diff
@@ -23,7 +23,15 @@ namespace starrocks::lake {
 template <>
 StatusOr<TabletMetadataPtr> MetadataIterator<TabletMetadataPtr>::get_metadata_from_tablet_manager(
         const std::string& path) {
-    return _manager->get_tablet_metadata(path, false);
+    auto tablet_metadata = _manager->get_tablet_metadata(path, false);
+    if (!tablet_metadata.ok() || tablet_metadata.value()->id() == _tablet_id) {
+        return tablet_metadata;
+    }
+
+    // Handle tablet initial metadata
+    auto metadata = std::make_shared<TabletMetadata>(*tablet_metadata.value());
+    metadata->set_id(_tablet_id);
```

Comment:
looks like the tablet_id is already set by TabletManager::get_tablet_metadata(), so this is not necessary here?

Reply:
get_tablet_metadata() has 2 overloaded functions, one gets tablet metadata by tablet_id and version, another gets tablet metadata by path, the former will set tablet_id if it's a initial tablet metadata, the former not